src/pyats/contrib/creators/topology.py

```diff
@@ -0,0 +1,775 @@
+        # add the device name to the respective set
+        for entry in results:
+            if results[entry].result()[0]:
+                self.cdp_configured.add(entry)
+            if results[entry].result()[1]:
+                self.lldp_configured.add(entry)
+        log.info('Devices that had cdp configured: {}, '
+                 'Devices that had lldp configured: {}'.format(self.cdp_configured, self.lldp_configured))
+
+    def process_neigbor_data(self, testbed, device_list, ip_net):
+        '''
+            Takes a testbed and processes the cdp and lldp data of every 
+            device on the testbed that has not yet been visited 
+        '''
+        dev_to_test = []
+        # Filter designed to get interface type from interface name
+        # Ex: Ethernet0/2 becomes Ethernet
+        interface_filter = re.compile(r'.+?(?=\d)')
+
+        # if the device has not been visited add it to the list of devices
+        # to get data from and add it to set of devices that have been examined
+        for device in testbed.devices:
+            if device not in self.visited_devices:
+                self.visited_devices.add(device)
+                dev_to_test.append(testbed.devices[device])
+        
+        # use pcall to get cdp and lldp information for all accessible devices
+        result = pcall(self.get_neighbor_info,
+                       device = dev_to_test)
+        conn_dict = {}
+
+        # process the connection data retrieved from getting cdp and lldp neighbors 
+        # and write it into a dictionary of format 
+        # {device:{interface with connection:{'dest_host': destination device,
+        #                                     'dest_port': destination device port, 
+        #                                     'ip_address': ip address of connection found}}} 
+        for entry in result:
+            for device in entry:
+                conn_dict[device] = self.get_device_connections(entry[device], 
+                                                                device, 
+                                                                device_list, 
+                                                                ip_net, 
+                                                                testbed)
+
+            # if device being visited doesn't have a given interface, 
+            # add the interface
+            for interface in conn_dict[device]:
+                if interface not in testbed.devices[device].interfaces:
+                    type_name = interface_filter.match(interface)
+                    interface_a = Interface(interface,
+                                            type = type_name[0].lower())
+                    interface_a.device = testbed.devices[device]
+        return conn_dict
+
+    def _process_cdp_information(self, result, entry, device_list, ip_net, testbed, connection_dict):
+        '''
+            Process the cdp parser information and enters it into the 

```

Comment:
updated `docstring` to have explanation for all arguments. please apply to all functions where applicable.

Reply:
Done